transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessagingClientImpl.kt

```diff
@@ -328,6 +337,19 @@ internal class MessagingClientImpl(
             is ErrorCode.RequestRateTooHigh,
             is ErrorCode.CustomAttributeSizeTooLarge,
             -> {
+                if (code is ErrorCode.CustomAttributeSizeTooLarge) {
+                    messageStore.clearInitialCustomAttributes()
+                    if (sendingAutostart) {
+                        sendingAutostart = false
+                        eventHandler.onEvent(
```

Comment:
In case of `autostart` message failure due to CA, you've mentioned that the `autostart` message will be re-sent.
How is it done? The eventListener is notified of the customAttributes error, in this case. On what error channel are you sending CA errors, is it both on messageListener and eventListener? Will the same error be caught twice if listening on both listeners?

Reply:
When CA exceed allowed size (that is basically the only reason why CA can fail) they fail the entire request. 

In case when CA attached to Message, they will fail the message request and this error will be communicated as error from within the messageListener. 

In case when CA attached to Autostart they will fail the Autostart request and the error will be comunicated as error on eventListener. 

It is not going to be duplicated on both channels. Only one of them will be reported.

There is an existing ticket on Shyrka to improve the overall UX. I have requested to add CA size limitation to SessionResponse. So if we know the CA limit ahead of sending the request and we find out that CA exceed this limitation instead of sending the request with them (and as a result failure the whole request) I will send an error event (via eventListener) stating that CA exceed max size and strip them down from the request (either Autostart or OnMessage). In that case request will be dispatched without CA, but developer also will be notified that they do not match the size requirements. 

here is a ticket: https://inindca.atlassian.net/browse/SHYRKA-2182